File: SwrveSDK/SDK/Messaging/SwrveCalibration.m
#import "SwrveCalibration.h"

@implementation SwrveCalibration

@synthesize renderScale;
@synthesize calibrationWidth;
@synthesize calibrationHeight;
@synthesize calibrationFontSize;
@synthesize calibrationText;

- (instancetype)initWithDictionary:(NSDictionary *)calibration {
    self = [super init];
    if (self) {
        self.calibrationWidth = (CGFloat)[[calibration objectForKey:@"width"] doubleValue];
        self.calibrationHeight = (CGFloat)[[calibration objectForKey:@"height"] doubleValue];
        self.calibrationFontSize = (CGFloat)[[calibration objectForKey:@"base_font_size"] doubleValue];
        self.calibrationText = [calibration objectForKey:@"text"];
    }
    return self;
}

@end

File: SwrveSDKSwift/SwrveCalibration.swift
import UIKit

@objc public class SwrveCalibration: NSObject {
    @objc public var calibrationWidth: CGFloat
    @objc public var calibrationHeight: CGFloat
    @objc public var calibrationFontSize: CGFloat
    @objc public var calibrationText: String

    @objc(initWithDictionary:)
    public init(calibration: [String: Any]) {
        self.calibrationWidth = CGFloat((calibration["width"] as? Double) ?? 0.0)
        self.calibrationHeight = CGFloat((calibration["height"] as? Double) ?? 0.0)
        self.calibrationFontSize = CGFloat((calibration["base_font_size"] as? Double) ?? 0.0)
        self.calibrationText = calibration["text"] as? String ?? ""
        super.init()
    }
}
